packages/docs/docs/advanced/canvas-operations.mdx

```diff
@@ -0,0 +1,110 @@
+`opacity`, can be set directly on the [`Node`](/api/2d/components/Node#opacity),
+and as a result are not exported from [partials](/api/2d/partials#Functions).
+
+`filters` accepts an array. This means that you can stack multiple filters onto
+a single Node. The order in which you add them does matter, as can be seen
+below:
+
+<AnimationPlayer name="canvas-operations-filters-stacking" />
+
+You can see that for example the green trail turns yellow on the right side
+while remaining green on the left.
+
+## Masking and Composite Operations
+
+Composite Operations define how two layers are stacked ontop of each other. This
+way it is possible to define Masks. MDN has a
+[great visualisation of all Composite Operations](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation#operations).
+
+If you are familiar with Adobe After Effects, you might know about
+[Track Mattes](https://helpx.adobe.com/after-effects/using/track-mattes-and-traveling-mattes.html).
+Track Mattes allow you to define a Background "Color" Layer, and a Foreground
+"Stencil" Layer. The Alpha Matte Layer for example will multiply the
+background's color by the stencil's alpha value.
+
+The same effect can be achieved in Motion Canvas with the help of Composite
+Operations using `source-in` and `source-out`.
+
+Create a new cached [`Node`](/api/2d/components/Node) that contains 2 children,
+the "stencil", and the "value" layers.
+
+```tsx
+const yOffset = createSignal(0);
+yield scene.add(<Background/>);
+yield scene.add(
+  <Node cache>
+    {/** This is your "stencil". It determines if something is visible or not */}
+    <AnimatedMotionCanvasIcon ... />
+    {/** This is your "value". It determines the output color if the stencil doesn't block it */}
+    <Img y={yOffset} src={exampleImage}
+      {/* 'source-in' for alpha stencil, 'source-out' for inverted alpha stencil*/}
+      compositeOperation={'source-in'}
+    />
+  </Node>
+);
+
+yield* yOffset(100, 2);
+yield* yOffset(-100, 2);
+```
+
+<AnimationPlayer name="canvas-operations-alpha-matte" />
+An example of how composite operations are used to create a mask. The Motion Canvas
+Icon acts as the stencil, the image in the back as the value.
+
+## Cached Nodes
+
+Both Filters and Composite Operations require a cached
+[`Node`](/api/2d/components/Node). Filters can set it automatically, while
+Composite Operations require you to set it explicitly on the parent
+[`Node`](/api/2d/components/Node).
+
+A cached [`Node`](/api/2d/components/Node) and its children are rendered on an
+offscreen canvas first, before getting added to the main scene.  
+For filters this is needed because they are applied to the entire canvas. By
+creating a new canvas and moving the elements that should get affected by the
+filters over, applying filters to the entire "new" canvas, and then moving back
+the result, you effectively only apply the filters to the moved elements.
```

Comment:
> you effectively only apply the filters to the moved elements

That's not the reason we render stuff onto a separate canvas, though.
The filters are always applied *only* to whatever you are drawing.
But there's a difference between drawing two blurred nodes separately, one after the other. And drawing them first and then blurring the result.
This is especially noticeable when it comes to opacity. Which will also force the node to be `cached`.